Lib/defconQt/glyphView.py

```diff
@@ -1651,62 +2138,52 @@ def createAnchor(self):
-            else:
-                return
-        if not self._glyph.name in newLayer:
-            newLayer.newGlyph(self._glyph.name)
-            # TODO: generalize this out, can’t use newStandardGlyph unfortunately
-            newLayer[self._glyph.name].width = self._glyph.width
-            newLayer[self._glyph.name].template = True
-        newGlyph = newLayer[self._glyph.name]
-        self.setGlyph(newGlyph)
-
-    def setGlyph(self, glyph):
+    def _makeLayerGlyph(self, layer):
+        name = self._name
+        layer.newGlyph(name)
+        glyph = layer[name]
+        # TODO: generalize this out, can’t use newStandardGlyph unfortunately
+        glyph.width = self.defaultWidth
+
+        # What does this do actually?
+        glyph.template = True
+
+        # TODO: it would be nice if a glyph that is created here would
+        # be more like a notion that a glyph should be created. And
+        # unless there's no attempt to change/draw into that glyph, it
+        # would not really be created. That way, the layer would be free
+        # of just empty glyphs and nothing unneccessary would go to disk.
+        return glyph
+
+    def changeCurrentLayer(self, layer):
+        name = self._name
+
+        # set current layer
+        self._layer = layer
+
+        if name in layer:
+            # please redraw
+            self.redraw()
+        else:
+            # no need to redraw, will happen within _layerGlyphAdded
+            self._makeLayerGlyph(layer)
+
+        # TODO: Undo data does not keep track of different layers
         scene = self.scene()
-        self._glyph.removeObserver(self, "Glyph.Changed")
-        self._glyph.layerSet.removeObserver(self, "LayerSet.Changed")
-        # TODO: consider creating a new scene instead of zeroing things out
-        # manually
         scene._dataForUndo = []
         scene._dataForRedo = []
-        self._glyph = glyph
-        app = QApplication.instance()
-        app.setCurrentGlyph(glyph)
-        self._glyph.addObserver(self, "_glyphChanged", "Glyph.Changed")
-        self._glyph.layerSet.addObserver(self, "_layersChanged", "LayerSet.Changed")
-        self.parent().setWindowTitle(self._glyph.name, self._glyph.getParent())
-        self.redrawGlyph()
-        self.redrawOtherLayers()
 
     def showEvent(self, event):
         super(GlyphView, self).showEvent(event)
         font = self._glyph.getParent()
-        # TODO: we should have an app-wide mechanism to handle default metrics
-        # values (that are applied to new fonts as well)
-        descender = font.info.descender
-        if descender is None: descender = -250
-        unitsPerEm = font.info.unitsPerEm
-        if unitsPerEm is None: unitsPerEm = 1000
-        self.fitInView(0, descender, self._glyph.width, unitsPerEm, Qt.KeepAspectRatio)
+        self.fitInView(0, font.info.descender, self._glyph.width, font.info.unitsPerEm, Qt.KeepAspectRatio)
```

Comment:
Okay. Creating a new branch and cherry-picking commits is the best way to rebase in my opinion.